napari/_qt/qt_event_loop.py

```diff
@@ -1,38 +1,123 @@
+        Set app version (if creating for the first time), by default __version__
+    icon : str, optional
+        Set app icon (if creating for the first time), by default
+        NAPARI_ICON_PATH
+    org_name : str, optional
+        Set organization name (if creating for the first time), by default
+        'napari'
+    org_domain : str, optional
+        Set organization domain (if creating for the first time), by default
+        'napari.org'
+    app_id : str, optional
+        Set organization domain (if creating for the first time).  Will be
+        passed to set_app_id (which may also be called independently), by
+        default NAPARI_APP_ID
+
+    Returns
+    -------
+    QApplication
+        [description]
 
     Notes
     -----
-    Substitute QApplicationWithTracing when using perfmon.
+    Substitutes QApplicationWithTracing when the NAPARI_PERFMON env variable
+    is set.
 
-    With IPython/Jupyter we call convert_app_for_tracing() which deletes
-    the QApplication and creates a new one. However here with gui_qt we
-    need to create the correct QApplication up front, or we will crash.
-    We'll crash because we'd be deleting the QApplication after we created
-    QWidgets with it, such as we do for the splash screen.
+    If the QApplication already exists, we call convert_app_for_tracing() which
+    deletes the QApplication and creates a new one. However here with get_app
+    we need to create the correct QApplication up front, or we will crash
+    because we'd be deleting the QApplication after we created QWidgets with
+    it, such as we do for the splash screen.
     """
-    if perf_config and perf_config.trace_qt_events:
-        from .perf.qt_event_tracing import QApplicationWithTracing
 
-        return QApplicationWithTracing(argv)
+    app = QApplication.instance()
+    if app:
+        if perf_config and perf_config.trace_qt_events:
+            from .perf.qt_event_tracing import convert_app_for_tracing
+
+            # no-op if app is already a QApplicationWithTracing
+            app = convert_app_for_tracing(app)
+        app._existed = True
     else:
-        return QApplication(argv)
+        # automatically determine monitor DPI.
+        # Note: this MUST be set before the QApplication is instantiated
+        QApplication.setAttribute(Qt.AA_EnableHighDpiScaling)
+
+        if perf_config and perf_config.trace_qt_events:
+            from .perf.qt_event_tracing import QApplicationWithTracing
+
+            app = QApplicationWithTracing(sys.argv)
+        else:
+            app = QApplication(sys.argv)
+
+        # if this is the first time the Qt app is being instantiated, we set
+        # the name, so that we know whether to raise_ in Window.show()
+
+        app.setApplicationName(app_name)
+        app.setApplicationVersion(app_version)
+        app.setOrganizationName(org_name)
+        app.setOrganizationDomain(org_domain)
+        app.setWindowIcon(QIcon(icon))
```

Comment:
- that is by design.  If ipython created the application, then we _don't want_ to set those properties.
- this is how it with `applicationName` [already is in master](url)
- all of those other properties besides `applicationName` are new anyway, so they're not being set in master as it is.

Reply:
`app.setWindowIcon(QIcon(icon))` is called on all Viewer creation (if is not blocked by class variable) on master and only on creation application in this PR` 

https://github.com/napari/napari/blob/35c9fa19a58d0b6245cdfb186a2e085612dfc729/napari/_qt/qt_main_window.py#L125-L127

Reply:
Correct!

Reply:
(That's the point)

Reply:
I guess what I'm asking is, rather than pointing out what this PR is doing (which I'm aware of), can you tell us whether there's anything you're unable to do in your app with this